packages/peregrine/lib/talons/SearchBar/useAutocomplete.js

```diff
@@ -33,21 +34,26 @@ export const useAutocomplete = props => {
     // run the query once on mount, and again whenever state changes
     useEffect(() => {
         if (visible && valid) {
+            setCleared(false);
```

Comment:
This will actually still display the stale results if the clear and next keystroke is done within 500ms. I'm not exactly sure why, but I suspect that while the query is in flight, it might be clearing out the contents of data, but haven't dug into it.

Very minor problem, but if you wanted to fix, you could use the `onCompleted` callback of `useLazyQuery`, or even easier just move the state update to `debouncedRunQuery`.

Reply:
> This will actually still display the stale results if the clear and next keystroke is done within 500ms.

I can't reproduce this. Are you encountering a situation in which the input is empty *and* the dropdown is open *and* stale results are displayed?

Stale results are always going to be present in the dropdown—that's the intent, given that I have no way to actually reset the query state—but this PR attempts to ensure that the dropdown is hidden (and thus, stale results are hidden) while the input is empty.

Reply:
I believe so. The effect immediately sets cleared to false, but doesn't account for the debounced delay. Very minor, but I think I captured it here.
![](https://gyazo.com/ee091f1474ea6ebaf7d3fbdce667a948)

Reply:
Right. As you can see in the gif, when the input is empty, the stale results are not shown. Instead, "Search for a product" is shown as a prompt. This is the new expected behavior.

The key thing I'm trying to communicate here is that we can't actually blow away the stale results of the query. Apollo gives us no method to do that. So we can only hide them, and I only want to hide them while the input is empty. 👍

Reply:
Fair enough. I think this debounced function provides a better experience; give it a try, and if you're still comfortable with the existing behavior, I'm fine with this as-is. Clean up the unit test failures and remove `location` propType in `SearchField` and I'll approve.

```js
const debouncedRunQuery = useMemo(
    () =>
        debounce(options => {
            setCleared(false);
            runQuery(options);
        }, 500),
    [runQuery]
);
```

Reply:
Good suggestion. I'll incorporate it.